flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/runner/AbstractDispatcherLeaderProcess.java

```diff
@@ -139,6 +139,8 @@ public final UUID getLeaderSessionId() {
 	private void closeInternal() {
 		log.info("Stopping {}.", getClass().getSimpleName());
 
+		state = State.STOPPED;
```

Comment:
Does this change belong into `cd14c5907bdbd88861f9e74448ded52d5c15a7a6`? (Because it requires other components to be stopped after the leader process transitioned to STOPPED.

Reply:
Yes I think it is correct. `State.STOPPED` effectively means that `closeInternal` has been called. But it does not say that the stopping is complete.